pkg/deploy/api/types.go

```diff
@@ -441,6 +441,20 @@ type DeploymentConfigRollbackSpec struct {
 	IncludeStrategy bool
 }
 
+// DeploymentRequest is a request to a deployment config for a new deployment.
+type DeploymentRequest struct {
+	unversioned.TypeMeta
+	// Name of the deployment config for requesting a new deployment.
+	Name string
+	// DryRun if true should not persist the updated deployment config in storage.
+	DryRun bool
```

Comment:
Since this is not required for the first version of `DeploymentRequest` I am going to remove it for now.